File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.Deque;
import java.util.LinkedList;

public class Solution {
    public int[] solution(int []arr) {
        Deque<Integer> queue = new LinkedList<>();
        
        for (int num : arr) {
            if (queue.isEmpty() || queue.peekLast() != num) {
                queue.add(num);
            }
        }
        
        int[] result = new int[queue.size()];
        int idx = 0;
        
        while (!queue.isEmpty()) {
            result[idx++] = queue.pollFirst();
        }
        
        return result;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr) {
    const stack = [];
    const result = []
    
    arr.forEach(num => {
        if (num === stack[0]) {
            return;
        }
        stack.pop();
        stack.push(num);
        result.push(num);
    });
    
    return result;
}